mantaray/received.py

```diff
@@ -256,23 +256,42 @@ def _handle_authenticate(server_view: views.ServerView) -> None:
         server_view.core.send("AUTHENTICATE " + b64_query[i : i + 400])
 
 
+class _JoinInProgress:
+    def __init__(self) -> None:
+        self.topic: str | None = None
+        self.nicks: list[str] = []
+
+
+_joins_in_progress: dict[tuple[views.ServerView, str], _JoinInProgress] = {}
+
+
+def _handle_numeric_rpl_topic(server_view: views.ServerView, args: list[str]) -> None:
+    channel, topic = args[1:]
+    join = _joins_in_progress.setdefault(
+        (server_view, channel.lower()), _JoinInProgress()
```

Comment:
We can assume the server is reasonable enough to send channel names cased consistently, no need to `.lower()`